File: NC/is-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        start = 0
        end = len(s)-1

        while start < end:
            if not self.alphaNum(s[start]):
                start += 1
                continue

            if not self.alphaNum(s[end]):
                end -= 1
                continue

            if s[start].lower() != s[end].lower():
                return False

            start += 1
            end -= 1

        return True

    def alphaNum(self, c):
        return (ord('A') <= ord(c) <= ord('Z') or
                ord('a') <= ord(c) <= ord('z') or
                ord('0') <= ord(c) <= ord('9'))

File: NC/is-palindrome.cpp
class Solution
{
public:
    bool isPalindrome(string s)
    {
        for (int i = 0, j = s.length() - 1; i < j;)
        {
            if (!((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z') ||
                  (s[i] >= '0' && s[i] <= '9')))
            {
                i++;
                continue;
            }
            if (!((s[j] >= 'a' && s[j] <= 'z') || (s[j] >= 'A' && s[j] <= 'Z') ||
                  (s[j] >= '0' && s[j] <= '9')))
            {
                j--;
                continue;
            }
            if (tolower(s[i]) != tolower(s[j]))
            {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};
